src/idfxt/JSONDataInterface.h

```diff
@@ -0,0 +1,102 @@
+// copyright 2014, Shannon Mackey <mackey@BUILDlab.net>
+#ifndef JSONDATAINTERFACE_H
+#define JSONDATAINTERFACE_H
+
+//for JSONDataObject and JSONDataInterface
+#include <string>
+#include <list>
+#include <map>
+#include <memory>
+
+//for old IDF text import only
+#include <algorithm>
+#include <cctype>
+#include <cstring>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <vector>
+#include <chrono>
+#include <random>
+
+const std::string ACCEPTED_VERSION = "8.2";
+//end old IDF text import only
+
+extern "C"
+{
+#include "../../third_party/cJSON/cJSON.h"
+}
+
+namespace idfx {
+
+
+class JSONDataObject
+{
+public:
+    JSONDataObject(const std::string &json_content);
+    ~JSONDataObject();
+
+    std::map<std::string, std::string> getProperties();
+    std::string getPropertyValue(std::string);
+//    void setPropertyValue(std::string, std::string);
+//    void setPropertyValue(std::string, int);
+//    void setPropertyValue(std::string, double);
+
+    std::list<std::shared_ptr<JSONDataObject> > getExtensions();
```

Comment:
Extension objects (each with one single set of fields - those that were repeated in the old idf) are nested right inside the main extensible object.  The extension object Type names are the ones that end with "_x" and they never appear outside of their parent object, ie. never at the root of the file.  They'd never be referenced from outside of its own parent.  These extension objects are in an ordered array, that you get a handle on, with getExtensions(), that you can iterate through if you wish.  ...All good stuff for me to put in the README.

For those base objects that you have a reference (uuid from the data of a reference field) is the case that remains for a getByUuid() function. (I had forgotten, in these later incarnations of idfx where I cast off any speculative methods).  Currently, the objects are returned from getObjects() as a std::list.  I will change this to a std::map of all objects keyed by Uuid - more in line with the JSON itself, thereby delivering the ability to get an object by uuid directly (*not* _re-getting_ it from the JSON), but rather, using std::map's find() method that will then return the object that you already have based on the uuid for lookup.